jupyterhub_moss/utils.py

```diff
@@ -18,3 +20,20 @@ def find(function: Callable[[Any], bool], iterable: Iterable[Any]) -> Optional[A
         if function(item):
             return item
     return None
+
+_TIMELIMIT_REGEXP = re.compile(
+    "^(?:^(?P<days>[0-9]+)-)?(?P<hours>[0-9]+)(?::(?P<minutes>[0-5]?[0-9]))?(?::(?P<seconds>[0-5]?[0-9]))?$"
+)
+
+
+def parse_timelimit(timelimit: str) -> Optional[datetime.timedelta]:
+    """Parse a SLURM timelimit/walltime string.
+
+    Returns a timedelta or None if parsing failed.
```

Comment:
On second thought, perhaps it would be clearer if this function raised an error in case the `match is None` ?

This means no further test is needed when parsing the `options` and that it could be handled with a `try`/`except` when parsing the `sinfo` command ?

Reply:
I've done it.